### PR TITLE
[Scheduler] Add DateIntervalTrigger and DatePeriodTrigger

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Tickets       | n/a
| License       | MIT
| Doc PR        | -

The `PeriodicalTrigger` should only be used for a fixed period of time that can be converted to a number of seconds.
Using it for `DatePeriod` does not make sense as the number of seconds between 2 runs can differ.
This PR introduces 2 new triggers that behave correctly in such cases.

### CODE REVIEW

**Review by nicolas-grekas — COMMENTED** [review]
Is it worth keep PeriodicalTrigger then? Can't we create one using the new trigger?

**Comment by fabpot** [comment]
> Is it worth keep PeriodicalTrigger then? Can't we create one using the new trigger?

Maybe not.
@upyx Can you chime in here?

**Comment by fabpot** [comment]
I've removed `PeriodicalTrigger` and ported some missing features in `DateIntervalTrigger`, so it supports the same scope.

File: src/Symfony/Component/Scheduler/Trigger/DateIntervalTrigger.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            $from = new \DateTimeImmutable($from);
+        }
+        if (\is_string($until)) {
+            $until = new \DateTimeImmutable($until);
+        }
+        try {
+            if (is_int($interval)) {
+                $interval = new \DateInterval('PT'.$interval.'S');
+            } elseif (is_string($interval)) {
+                if ('P' === $interval[0]) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Scheduler/Trigger/DateIntervalTrigger.php]: if ('P' === ($interval[0] ?? '')) {

File: src/Symfony/Component/Scheduler/Trigger/DateIntervalTrigger.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    $interval = new \DateInterval($interval);
+                } elseif (ctype_digit($interval)) {
+                    $interval = new \DateInterval('PT'.$interval.'S');
+                } else {
+                    $interval = \DateInterval::createFromDateString($interval);
+                }
+            }
+        } catch (\Exception $e) {
+            throw new InvalidArgumentException(sprintf('Invalid interval "%s".', $interval), 0, $e);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Scheduler/Trigger/DateIntervalTrigger.php]: `sprintf('Invalid interval "%s":', $interval).$e->getMessage()`?

Comment (fabpot) [reply src/Symfony/Component/Scheduler/Trigger/DateIntervalTrigger.php]: done

File: src/Symfony/Component/Scheduler/Trigger/DateIntervalTrigger.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Scheduler/Trigger/DateIntervalTrigger.php]: yep

**Review by nicolas-grekas — APPROVED** [review]

**Comment by upyx** [comment]
@fabpot It doesn't work in real cases. Period iterator iterates over the all time points from the begginng. For example, some create a period from "2023-03-21" on every second. At "2024-03-12" the period will iterate over 31536000 poins on every worker start. It is the reason why I wrote different version.

Revert it, please.

**Comment by fabpot** [comment]
@upyx I'm on it.

**Pull request merged**